### PR TITLE
tiproxy: add performance report of traffic capture

### PR DESCRIPTION
**Pull request by djshow832**
### What is changed, added or deleted? (Required)

- Some community users gave feedback that they wanted the performance effect of traffic capture, so I did a performance test.
- Meanwhile, a colleague wanted to see the sample of `fail`, so I added a sample.
- Added more warnings as the PM required.
- Fix wrong name of db `tiproxy_traffic_replay`. 

### Which TiDB version(s) do your changes apply to? (Required)

**Tips for choosing the affected version(s):**

By default, **CHOOSE MASTER ONLY** so your changes will be applied to the next TiDB major or minor releases. If your PR involves a product feature behavior change or a compatibility change, **CHOOSE THE AFFECTED RELEASE BRANCH(ES) AND MASTER**.

For details, see [tips for choosing the affected versions (in Chinese)](https://github.com/pingcap/docs-cn/blob/master/CONTRIBUTING.md#版本选择指南).

- [x] master (the latest development version)
- [x] v8.4 (TiDB 8.4 versions)
- [ ] v8.3 (TiDB 8.3 versions)
- [ ] v8.2 (TiDB 8.2 versions)
- [ ] v8.1 (TiDB 8.1 versions)
- [ ] v7.5 (TiDB 7.5 versions)
- [ ] v7.1 (TiDB 7.1 versions)
- [ ] v6.5 (TiDB 6.5 versions)
- [ ] v6.1 (TiDB 6.1 versions)
- [ ] v5.4 (TiDB 5.4 versions)
- [ ] v5.3 (TiDB 5.3 versions)

### What is the related PR or file link(s)?

- This PR is translated from:
- Other reference link(s):

### Do your changes match any of the following descriptions?

- [ ] Delete files
- [ ] Change aliases
- [ ] Need modification after applied to another branch 
- [ ] Might cause conflicts after applied to another branch

### CODE REVIEW

File: tiproxy/tiproxy-traffic-replay.md [hunk 1]
```diff
@@ -76,7 +77,7 @@ summary: 介绍 TiProxy 的流量回放的使用场景和使用步骤。
 
 5. 查看回放报告。
 
-    回放完成后，报告存储在测试集群的 `tiproxy_traffic_report` 数据库下。该数据库包含两个表 `fail` 和 `other_errors`。
+    回放完成后，报告存储在测试集群的 `tiproxy_traffic_replay` 数据库下。该数据库包含两个表 `fail` 和 `other_errors`。
```
Comment (joey-yez) [inline tiproxy/tiproxy-traffic-replay.md:80]: 如果测试集群已经存在 这个 数据库，或者多次回放时，数据回放后的表现，最好也在这里介绍一下。

**Review by joey-yez — APPROVED** [review]

File: tiproxy/tiproxy-performance-test.md [hunk 2]
```diff
@@ -14,6 +14,7 @@ summary: TiProxy 的性能测试报告、与 HAProxy 的性能对比。
 - 查询结果集的行数对 TiProxy 的 QPS 有显著影响，且影响程度与 HAProxy 相同
 - TiProxy 的性能随 vCPU 的数量接近线性增长，因此增加 vCPU 的数量可以有效提高 QPS 上限
 - 长连接的数量、短连接的创建频率对 TiProxy 的 QPS 影响很小
+- TiProxy 的 CPU 使用率越高，捕获流量对 QPS 的影响越大。在 TiProxy 的 CPU 使用率约 70% 时，开启流量捕获使平均 QPS 下降约 3%，同时最低 QPS 下降约 7%，后者是由压缩流量文件导致 QPS 周期性下降。
```
Comment (Oreoxmt) [inline tiproxy/tiproxy-performance-test.md:17]:
```suggestion
- TiProxy 的 CPU 使用率越高，开启捕获流量功能对 QPS 的影响越大。当 TiProxy 的 CPU 使用率约为 70% 时，开启流量捕获会导致平均 QPS 下降约 3%，最低 QPS 下降约 7%，后者的下降是由压缩流量文件导致的 QPS 周期性下降。
```

File: tiproxy/tiproxy-performance-test.md [hunk 3]
```diff
@@ -312,3 +313,35 @@ sysbench oltp_point_select \
 | 100 | 95597 | 0.52     | 0.65       | 330% | 1800% |
 | 200 | 94692 | 0.53     | 0.67       | 330% | 1800% |
 | 300 | 94102 | 0.53     | 0.68       | 330% | 1900% |
+
+## 捕获流量测试
+
+### 测试方案
+
+该测试的目的是测试捕获流量对性能的影响。该测试使用 TiProxy v1.3.0 版本，在执行 `sysbench` 之前分别关闭和开启流量捕获，同时调整并发度，对比 QPS 和 TiProxy 的 CPU 使用率。由于周期性的压缩文件会引起 QPS 的波动，因此本测试除了对比平均 QPS，还对比最低 QPS。
```
Comment (Oreoxmt) [inline tiproxy/tiproxy-performance-test.md:321]: > 该测试使用 TiProxy v1.3.0 版本

与文档开头 测试环境 > 软件版本 部分不一致
https://github.com/pingcap/docs-cn/blob/c4a38f00313b83ece1e554b0b85923df73e2fbbb/tiproxy/tiproxy-performance-test.md?plain=1#L32-L36

Comment (Oreoxmt) [inline tiproxy/tiproxy-performance-test.md:321]:
```suggestion
该测试的目的是测试[捕获流量](/tiproxy/tiproxy-traffic-replay.md)对 TiProxy 性能的影响。该测试使用 TiProxy v1.3.0 版本，在执行 `sysbench` 之前分别关闭和开启流量捕获，同时调整并发度，以对比 QPS 和 TiProxy 的 CPU 使用率。由于周期性的压缩文件会引起 QPS 的波动，因此本测试除了对比平均 QPS，还对比最低 QPS。
```

File: tiproxy/tiproxy-traffic-replay.md [hunk 4]
```diff
@@ -89,16 +91,50 @@ summary: 介绍 TiProxy 的流量回放的使用场景和使用步骤。
...
+        sample_err_msg: this is an error from the backend connection: failed to read the connection: EOF
+    sample_replay_time: 2024-10-17 12:57:39
+                 count: 1
+    ```
+
     > **注意：**
     >
-    > `tiproxy_traffic_report` 中的表结构在未来版本中可能会改变。不推荐在应用程序开发或工具开发中读取 `tiproxy_traffic_report` 中的数据。
+    > - `tiproxy_traffic_replay` 中的表结构在未来版本中可能会改变。不推荐在应用程序开发或工具开发中读取 `tiproxy_traffic_replay` 中的数据。
+    > - 回放不保证连接之间的事务执行顺序与捕获时完全一致，因此可能会产生误报。
```
Comment (Oreoxmt) [inline tiproxy/tiproxy-traffic-replay.md:136]:
```suggestion
    > - 回放不保证连接之间的事务执行顺序与捕获时完全一致，因此可能会产生误报错误。
```

File: tiproxy/tiproxy-performance-test.md (hunk 3 above)
Comment (djshow832) [reply tiproxy/tiproxy-performance-test.md:321]: 是的，我没必要全部用 v1.3.0 再测一次，代价太大了

File: tiproxy/tiproxy-traffic-replay.md (hunk 4 above)
Comment (djshow832) [reply tiproxy/tiproxy-traffic-replay.md:136]: 要么 "因此可能会误报错误"？"产生" 和 "误报" 这里都是动词。

**Review by qiancai — APPROVED** [review]
Rest LGTM

**Review by Oreoxmt — APPROVED** [review]

**Pull request merged**